sixtyfps_compiler/generator.rs

```diff
@@ -143,51 +143,63 @@ pub fn build_array_helper(component: &Component, mut visit_item: impl FnMut(&Ele
     }
 }
 
+/// A reference to this trait is passed to the [`build_item_tree`] function.
+/// It can be used to build the array for the item tree.
+pub trait ItemTreeBuilder {
+    /// Some state that contains the code on how to access some particular component
+    type SubComponentStack;
```

Comment:
In the two implementations this associated type is either a `String` or a new-typed `int` (the item index), neither of which I think are stacks or treated as such. I'd prefer "State" instead of "Stack".